luigi_pipeline/seqr_mito_loading.py

```diff
@@ -0,0 +1,109 @@
+        pprint.pprint(self.__dict__)
+
+        mt = hl.read_matrix_table(self.source_paths[0])
+        mt = mt.drop('dbsnp_version',
+            'dp_hist_all_variants_n_larger', 'mq_hist_all_variants_bin_edges', 'dp_hist_all_variants_n_larger',
+            'age_hist_all_samples_n_smaller', 'age_hist_all_samples_bin_freq', 'age_hist_all_samples_n_larger',
+            'tlod_hist_all_variants_bin_freq', 'dp_hist_all_variants_bin_freq', 'mq_hist_all_variants_bin_freq',
+            'col_annotation_descriptions', 'dp_hist_all_variants_bin_edges', 'mq_hist_all_variants_n_larger',
+            'age_hist_all_samples_bin_edges', 'tlod_hist_all_variants_bin_edges', 'pop_order', 'hap_order',
+            'global_annotation_descriptions', 'row_annotation_descriptions', 'tlod_hist_all_variants_n_larger')
+        if not self.dont_validate:
+            self.validate_mt(mt, self.genome_version, self.sample_type)
+        if self.remap_path:
+            mt = self.remap_sample_ids(mt, self.remap_path)
+        if self.subset_path:
+            mt = self.subset_samples_and_variants(mt, self.subset_path)
+
+        ref_data = hl.read_table(self.reference_ht_path)
+
+        high_constraint_region = hl.import_locus_intervals(self.high_constraint_interval_path, reference_genome='GRCh38')
+
+        mt = schema_cls(mt, ref_data=ref_data, high_constraint_region=high_constraint_region).annotate_all(
+            overwrite=True).select_annotated_mt()
+
+        mt = mt.annotate_globals(sourceFilePath=','.join(self.source_paths),
+                                 genomeVersion=self.genome_version,
+                                 sampleType=self.sample_type,
+                                 hail_version=pkg_resources.get_distribution('hail').version)
+
+        mt.describe()
+        mt.write(self.output().path, stage_locally=True, overwrite=True)
+
+    def run(self):
+        # We only want to use the Variant Schema.
+        self.read_mt_write_mt(schema_cls=SeqrMitoVariantSchema)
+
+
+class SeqrMitoGenotypesMTTask(HailMatrixTableTask):
+    remap_path = luigi.OptionalParameter(default=None,
+                                         description="Path to a tsv file with two columns: s and seqr_id.")
+    subset_path = luigi.OptionalParameter(default=None,
+                                          description="Path to a tsv file with one column of sample IDs: s.")
+
+    def requires(self):
+        return [SeqrMitoVariantMTTask()]
+
+    def run(self):
+        mt = hl.read_matrix_table(self.input()[0].path)
+
+        if self.remap_path:
+            mt = self.remap_sample_ids(mt, self.remap_path)
+        if self.subset_path:
+            mt = self.subset_samples_and_variants(mt, self.subset_path)
+
+        mt = SeqrMitoGenotypesSchema(mt).annotate_all(overwrite=True).select_annotated_mt()
+
+        mt.describe()
+        mt.write(self.output().path, stage_locally=True, overwrite=True)
+
+
+class SeqrMTToESOptimizedTask(HailElasticSearchTask):
```

Comment:
this looks identical to `SeqrMTToESOptimizedTask` except for a couple different class names. Whenever you find yourself copy-pasting large amounts of code, it almost certainly means you should be using inheritance instead. Define a base class and inherit as needed:
```
class BaseMTToESOptimizedTask(HailElasticSearchTask):

   ...
    def requires(self):
        return [self.VCF_TO_VARIANT_TASK(), self.VCF_TO_GENOTYPES_TASK()]
    ...
    def run(self):
        ...
        row_ht = self.VARIANT_GENOTYPES_SCHEMA.elasticsearch_row(row_ht)

...
class SeqrMTToESOptimizedTask(BaseMTToESOptimizedTask):
    VCF_TO_VARIANT_TASK = SeqrMitoVariantMTTask
    VCF_TO_GENOTYPES_TASK = SeqrMitoGenotypesMTTask
    VARIANT_GENOTYPES_SCHEMA = SeqrMitoVariantsAndGenotypesSchema
```